web/client/api/WFS.js

```diff
@@ -5,48 +5,100 @@
-            }
-            return response.data;
-        });
-    },
-    describeFeatureType: function(url, typeName) {
-        const parsed = urlUtil.parse(url, true);
-        const describeLayerUrl = urlUtil.format(assign({}, parsed, {
-            query: assign({
+import axios from '../libs/ajax';
+import urlUtil from 'url';
+import assign from 'object-assign';
+
+const toDescribeURL = (url, typeName) => {
+    const parsed = urlUtil.parse(url, true);
+    return urlUtil.format(
+        {
+            ...parsed,
+            search: undefined, // this allows to merge parameters correctly
+            query: {
+                ...parsed.query,
+
                 service: "WFS",
                 version: "1.1.0",
-                typeName: typeName,
+                typeName,
+                outputFormat: 'application/json',
                 request: "DescribeFeatureType"
-            }, parsed.query)
-        }));
-        return new Promise((resolve) => {
-            require.ensure(['../utils/ogc/WFS'], () => {
-                const {unmarshaller} = require('../utils/ogc/WFS');
-                resolve(axios.get(describeLayerUrl).then((response) => {
-                    let json = unmarshaller.unmarshalString(response.data);
-                    return json && json.value;
-
-                }));
-            });
+            }
         });
-    }
+};
+/**
+ * Simple getFeature using http GET method with json format
+ */
+export const getFeatureSimple = function(baseUrl, params) {
+    return axios.get(baseUrl + '?service=WFS&version=1.1.0&request=GetFeature', {
+        params: assign({
+            outputFormat: "application/json"
+        }, params)
+    }).then((response) => {
+        if (typeof response.data !== 'object') {
+            return JSON.parse(response.data);
+        }
+        return response.data;
+    });
+};
+
+export const getCapabilitiesURL = (url, {version = "1.1.0"} = {}) => {
+    const parsed = urlUtil.parse(url, true);
+    return urlUtil.format(assign({}, parsed, {
+        query: assign({
+            service: "WFS",
+            version,
+            request: "GetCapabilities"
+        }, parsed.query)
+    }));
+};
+
+export const getFeatureURL = (url, typeName, { version = "1.1.0", ...params } = {}) => {
+    const parsed = urlUtil.parse(url, true);
+    return urlUtil.format(assign({}, parsed, {
+        query: assign({
+            service: "WFS",
+            typeName,
+            version,
+            request: "GetFeature",
+            ...params
+        }, parsed.query)
+    }));
+};
+
+export const getFeature = (url, typeName, params ) => {
+    return axios.get(getFeatureURL(url, typeName, params));
+};
+
+export const getCapabilities = function(url) {
+    return axios.get(getCapabilitiesURL(url));
+};
+/**
+ * @deprecated
+ */
+export const describeFeatureTypeOGCSchemas = function(url, typeName) {
+    const parsed = urlUtil.parse(url, true);
```

Comment:
Why are we adding more stuff depending on ogc-schemas?

Reply:
No, here I changed the name of the old describeFeatureType that uses OGC schemas method and deprecated it.

Reply:
fine